### PR TITLE
Modify CLI client to accept socks proxy (fixes #2289)

### LINKED ISSUE #2289: Add option to CLI to configure socks proxy
The HttpClient in com.facebook.presto.cli.QueryRunner uses a hard-coded config with no socks proxy support. We need to make this configurable via a command line option to the CLI.

### PR DESCRIPTION
**Pull request by kaschaeffer**
cc @martint

### CODE REVIEW

**Comment by kaschaeffer** [comment]
Fixes #2289.  The CI errors seem unrelated to this PR -- anything I need to fix there?

**Comment by kaschaeffer** [comment]
@martint I agree about "socks-proxy" -- I chose "socks" for compatibility with [BenchmarkDriverOptions](https://github.com/facebook/presto/blob/0ab67c7c36ce5157f8f56537839cca04bc791004/presto-benchmark-driver/src/main/java/com/facebook/presto/benchmark/driver/BenchmarkDriverOptions.java#L79) but can change

**Comment by electrum** [comment]
@kaschaeffer good point about benchmark-driver. We should update that too.

File: presto-cli/src/main/java/com/facebook/presto/cli/Console.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import com.google.common.base.Throwables;
 import com.google.common.collect.ImmutableSet;
 import com.google.common.io.Files;
+import com.google.common.net.HostAndPort;
```
Comment (electrum) [inline presto-cli/src/main/java/com/facebook/presto/cli/Console.java]: Unused import. The IDE will fix these: Auto Import -> Optimize imports on the fly

**Comment by electrum** [comment]
I made those minor fixes and merged. Thanks!

**Comment by nezihyigitbasi** [comment]
@electrum I guess airlift ``JettyHttpClient`` only supports socks4 so does this PR, any plans to support socks5?

**Comment by dain** [comment]
@nezihyigitbasi last time I checked it only supported sock4. do you need socks5?  It should be pretty easy to add.  I just added support in the test suite (I was playing with OkHttp which only supports socks5).

**Comment by nezihyigitbasi** [comment]
@dain not that urgent, but nice to have. Currently jetty only supports socks4, socks5 has to be implemented. Btw can you point me to the test suite you are talking about?

**Comment by dain** [comment]
https://github.com/dain/airlift/commit/39ff0d8e1e9bc23037f23a88cecf87381f3f6929

**Pull request closed without merging**